### PR TITLE
Create improved algorithm for pulse time

### PR DESCRIPTION
**Pull request by watsonjj**
We utilise robust peak finding algorithms in the charge extraction. We should use them in the pulse time extraction as well.

This PR creates the `extract_pulse_time_around_peak` numba ufunc, which calculates the weighted average of the pulse within the window defined by the `ImageExtractor`.

### CODE REVIEW

File: ctapipe/image/extractor.py [hunk 1]
```diff
@@ -124,11 +150,19 @@ def extract_pulse_time_weighted_average(waveforms):
...
+
+    num = 0
+    den = 0
+    for isample in prange(start, end):
+        if 0 <= isample < n_samples:
+            num += waveforms[isample] * isample
+            den += waveforms[isample]
+
+    # TODO: Return pulse time in units of ns instead of isample
+    ret[0] = num / den if den > 0 else peak_index
```
Comment (kosack) [inline ctapipe/image/extractor.py]: I think you might be missing a `return` statement here

Comment (kosack) [reply ctapipe/image/extractor.py]: that will also fix the errors in Codacy (since it complains about this)

Comment (watsonjj) [reply ctapipe/image/extractor.py]: No, this is how `@guvectorize` works, the return value is stored in the `ret` array

Comment (watsonjj) [reply ctapipe/image/extractor.py]: Like filling a vector in c++ that was passed to a function by reference

Comment (kosack) [reply ctapipe/image/extractor.py]: ah, I see - then we can ignore the codacy errors

**Pull request merged**